hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/FSHLog.java

```diff
@@ -1120,8 +1120,18 @@ public void shutdown() throws IOException {
           this.disruptor.shutdown();
         }
       }
-      // With disruptor down, this is safe to let go.
-      if (this.appendExecutor !=  null) this.appendExecutor.shutdown();
+
+      try {
+        // With disruptor down, this is safe to let go.
+        if (this.appendExecutor != null) {
+          this.appendExecutor.shutdown();
+          // Wait for 60 seconds to complete currently running tasks.
+          this.appendExecutor.awaitTermination(60, TimeUnit.SECONDS);
```

Comment:
Although 60s looks reasonable, want to make it configurable? I know another config is pain, but if the time comes, we can tweak it?

Reply:
Don't have a specific preference on hardcoded/configurability, whatever works for everyone.

OTOH, mind handling the return value of awaitTermination()? something like..

> if (!appendExecutor.awaitTermination(...)) {
>   LOG.error("Error terminating append pool, potential case for aborted append operations...)
> }

Reply:
> Although 60s looks reasonable, want to make it configurable? I know another config is pain, but if the time comes, we can tweak it?

@virajjasani  Thank you for the feedback. Added new config key `hbase.regionserver.wal.shutdown.timeout`. Default value to `hbase.regionserver.wal.sync.timeout`.

Reply:
> OTOH, mind handling the return value of awaitTermination()? something like..

@bharathv  Done ! please review again. Thank you !